python/embeddings.py

```diff
@@ -0,0 +1,136 @@
+import os
+import sys
+import multiprocessing as mp
+from concurrent.futures import ProcessPoolExecutor, as_completed
+import mgp
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "embed_worker"))
+
+logger: mgp.Logger = mgp.Logger()
+
+EXCLUDE_PROPERTIES = {"embedding"}
+BATCH_SIZE = 2000
+MODEL_ID = "all-MiniLM-L6-v2"
+
+
+def build_texts(vertices):
+    out = []
+    for v in vertices:
+        txt = " ".join(lbl.name for lbl in v.labels) + " " + " ".join(
+            f"{k}: {val}" for k, val in v.properties.items() if k not in EXCLUDE_PROPERTIES
+        )
+        out.append(txt)
+    return out
+
+
+def split_slices(n_items: int, n_parts: int):
+    base, rem = divmod(n_items, n_parts)
+    start = 0
+    slices = []
+    for i in range(n_parts):
+        end = start + base + (1 if i < rem else 0)
+        slices.append((start, end))
+        start = end
+    return slices
+
+
+def get_visible_gpus():
+    # Avoid creating a CUDA context in the parent if possible
+    try:
+        import subprocess
+        out = subprocess.check_output(
+            ["nvidia-smi", "--query-gpu=index", "--format=csv,noheader"], text=True
+        )
+        return [int(x) for x in out.strip().splitlines() if x.strip()]
+    except Exception:
+        try:
+            import torch
+            return list(range(torch.cuda.device_count())) if torch.cuda.is_available() else []
+        except Exception:
+            return []
+
+
+@mgp.write_proc
+def compute_embeddings(ctx: mgp.ProcCtx) -> mgp.Record(success=bool):
```

Comment:
Add vertices as an argument here, if passed only that subset of vertices should be calculated